mono/metadata/marshal-ilgen.c

```diff
@@ -4681,6 +4681,112 @@ emit_marshal_asany_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 	return conv_arg;
 }
 
+static int
+emit_marshal_date_time_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
+					MonoMarshalSpec *spec,
+					int conv_arg, MonoType **conv_arg_type,
+					MarshalAction action)
+{
+	MonoType *date_time_type, *double_type, *double_ref_type;
+	MonoMethodBuilder *mb = m->mb;
+	static MonoMethod *to_oadate;
+	static MonoMethod *from_oadate;
+	MonoClass *date_time_class;
+	int pos = 0;
+
+	date_time_class = mono_class_get_date_time_class ();
+	double_type = m_class_get_byval_arg (mono_defaults.double_class);
+	double_ref_type = m_class_get_this_arg (mono_defaults.double_class);
+
+	if (!from_oadate)
+		from_oadate = get_method_nofail (date_time_class, "FromOADate", 1, 0);
+	g_assert (from_oadate);
+	if (!to_oadate)
+		to_oadate = get_method_nofail (date_time_class, "ToOADate", 0, 0);
```

Comment:
Missing barriers.

Reply:
Could you elaborate? This is copied from existing code of _ACTION_CONV_ cases.

Reply:
The existing code looks wrong too.
See https://github.com/mono/mono/pull/17822.

Reply:
I guess nevermind the barriers. I'll try to get around to it, but we skate by, and x86 is already ok.